app/src/main/java/org/septa/android/app/locationpicker/LineAwareLocationPickerTabActivityHandler.java

```diff
@@ -186,6 +192,14 @@ public void onClick(View view) {
                     intent.putExtra(Constants.TRANSIT_TYPE, transitType);
                     intent.putExtra(Constants.ROUTE_DIRECTION_MODEL, selectedRoute);
 
+                    if (NextToArriveResultsActivity.class.equals(targetClass)) {
```

Comment:
Comparing classes is not safe. Add a field to the constructor that will know where this is being used.